File: Wish.UI/src/app/models/Product/ProductResponse.ts
export class ProductResponse {
  resultType = 0
  errors = []
  data: Data = {
    id: 0,
    title: '',
    description: '',
    categoryId: 0,
    pictureImage: '',
    weight: 0,
    material:{
      id: 0,
      type: 0,
      title: ''
    },
    count: 0,
    price: 0,
    attributes:[]

  }
}

export interface Data {
  id: 0
  title: string
  description: string
  categoryId: number
  pictureImage: string
  weight: number
  material: Material
  count: number
  price: number
  attributes: Attribute[]
}

export interface Material {
  id: number
  type: number
  title: string
}

export interface Attribute {
  id: number
  productId: number
  title: string
  value: string
}

File: Wish-Jewelry-Shop/src/Wish.Application/Responses/Product/ProductResponse.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Wish.Domain.Constants;
using Wish.Domain.Entities;
using Attribute = Wish.Domain.Entities.Attribute;

namespace Wish.Application.Responses.Product
{
	public class ProductResponse
	{
		public long Id { get; set; }
		public string Title { get; set; }

		public string Description { get; set; }

		public long CategoryId { get; set; }

		public string PictureImage { get; set; }

		public double Weight { get; set; }

		public Material Material { get; set; }

		public long Count { get; set; }

		public decimal Price { get; set; }

		public virtual List<Attribute> Attributes { get; set; }
	}
}
